### PR TITLE
Themes: updated Banner, better copy, tailored to plans

### PR DESCRIPTION
**Pull request by folletto**
This PR brings a series of improvements to the existing banner in Themes:

1. It shows different banners to Free and Personal -vs- Premium.
2. Shows the banner also to lower-tier plans, not just Free.
3. Updated copy (I need a review on this, it's just a first draft)
4. Switches out the deprecated `UpgradeNudge` with `Banner`.

Current (shown on Free only):

screen shot 2017-04-13 at 16 13 53

This PR, Banner for sites on Free and Personal (copy has been updated, see below):

screen shot 2017-04-13 at 16 02 01

This PR, Banner for sites on Premium (copy has been updated, see below):

screen shot 2017-04-13 at 16 01 53

Note that changing the component also changes tracking from `calypso_upgrade_nudge_cta_click` to `calypso_banner_cta_click`. In both cases the `cta_name` will stay the same: `themes_custom_design`. We might want do differentiate the two banners tho? Advice is welcome.

### To test

1. Open Calypso Themes
2. Check the banner on a Free site (green banner)
3. Check the banner on a Personal site (green banner)
4. Check the banner on a Premium site (violet banner)
5. Check the banner on a Business site (no banner)

### CODE REVIEW

File: client/my-sites/themes/single-site-wpcom.jsx [hunk 1]
```diff
@@ -10,26 +11,41 @@ import CurrentTheme from 'my-sites/themes/current-theme';
...
+					plan={ PLAN_PREMIUM }
+					title={ translate( 'Access all our premium themes with our Premium and Business plans!' ) }
+					description={ translate( 'Get also advanced customization, more space, and video support.' ) }
 					event="themes_custom_design"
-					/>
+				/>
+				}
+				{ ( currentPlanSlug === PLAN_PREMIUM ) && <Banner
+					plan={ PLAN_BUSINESS }
+					title={ translate( 'Why not upgrade to Business?' ) }
```
Comment (a8ci18n) [inline client/my-sites/themes/single-site-wpcom.jsx]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'Upgrade to Business' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by michelleweber** [comment]
On the violet banner, I'd delete the "more" before SEO tools to avoid having "more" twice in the sentence so close together. If you want a qualifier there, maybe "advanced SEO tools"?

In the green banner, the "also" is in an odd place in the sentence -- you can either just delete that word, or rephrase to something like "Get advanced customization, more storage space, and video support along with all your new themes."

**Comment by folletto** [comment]
Agreed on both, updated:

screen shot 2017-04-13 at 18 39 24

screen shot 2017-04-13 at 18 39 31

Thanks! :D

File: client/my-sites/themes/single-site-wpcom.jsx [hunk 2]
```diff
@@ -39,3 +59,12 @@ export default connectOptions(
 		);
 	}
  );
+
+export default connect(
+	( state, { siteId } ) => {
+		const currentPlan = getCurrentPlan( state, siteId ) || null;
```
Comment (ockham) [inline client/my-sites/themes/single-site-wpcom.jsx]: What would happen without the `null` fallback?

Comment (folletto) [inline client/my-sites/themes/single-site-wpcom.jsx]: Nothing in particular, just at some points it was "undefined" and at other points was "null", so I preferred to push it over the line.

File: client/my-sites/themes/single-site-wpcom.jsx [hunk 3]
```diff
@@ -39,3 +59,12 @@ export default connectOptions(
...
+		return {
+			currentPlanSlug: currentPlan && currentPlan.productSlug,
```
Comment (ockham) [inline client/my-sites/themes/single-site-wpcom.jsx]: Use lodash's `get`:

```js
currentPlanSlug: get( currentPlan, 'productSlug', false )
```

(3rd arg is fallback -- use whatever seems appropriate)

**Comment by folletto** [comment]
Done:

* Remove redundant `|| null` enforcement.
* Used lodash `get`.

**Comment by davidakennedy** [comment]
The revised copy here reads well to me, and I'm excited to get a theme focused nudge in here.

**Comment by folletto** [comment]
I added two different trackings for the two banners:

* `themes_plans_free_personal`
* `themes_plans_premium`

So we can check how they perform individually.

**Comment by folletto** [comment]
The discussion progressed a bit and we decided for this PR to remove the nudge to Premium users, as it seems a bit off to nudge already paying customers. However this PR is already beneficial as: 

* It uses the correct design pattern `Banner`
* It has a better copy
* It would add Personal Plans to it
* Makes trivial to re-add a banner for others

So we can push this live and review that other decision.

**Pull request merged**